lucetc/src/lib.rs

```diff
@@ -380,7 +357,7 @@ impl Lucetc {
         let dir = tempfile::Builder::new().prefix("lucetc").tempdir()?;
         let objpath = dir.path().join("tmp.o");
         self.object_file(objpath.clone())?;
-        link_so(objpath, &self.target, &output)?;
+        link_so(objpath, self.builder.target_ref(), &output)?;
```

Comment:
You mentioned a similar thing with `heap_settings_mut`, I think the same thing applies here. As well. Having an extra method doesn't hurt us, especially if it's scoped with `pub(crate)`